src/client/components/App/index.js

```diff
@@ -59,10 +80,46 @@ App.propTypes = {
   list: PropTypes.array.isRequired,
   title: PropTypes.string.isRequired,
   language: PropTypes.array.isRequired,
+  searchValue: PropTypes.string.isRequired,
+  search: PropTypes.func.isRequired,
+  filter: PropTypes.func.isRequired,
+  facetedSearch: PropTypes.func.isRequired,
 };
 
 App.childContextTypes = {
   store: PropTypes.object.isRequired,
 };
 
-export default connect(App);
+const filterList = ({ list, searchValue, filterValue, facetedValue }) => {
+  let newList = list;
+  if (facetedValue && facetedValue !== 'All') {
+    newList = newList.filter((li) => (li.Categories === facetedValue));
+  }
+  if (!searchValue && (!filterValue || filterValue === 'All')) {
+    return newList;
+  } else if (searchValue && (!filterValue || filterValue === 'All')) {
+    return newList.filter((li) => (li.Name.toLowerCase().match(searchValue.toLowerCase())));
+  } else if (filterValue && searchValue) {
+    return newList.filter((li) => (li.Name.toLowerCase().match(searchValue.toLowerCase()) && filterValue === li.Type));
+  } else if (!searchValue && (filterValue && filterValue !== 'All')) {
+    return newList.filter((li) => (filterValue === li.Type));
+  }
+};
+
+const mapStateToProps = state => ({
+  title: state.title,
+  facetedbox: state.facetedbox,
+  filterbox: state.filterbox,
+  language: state.language,
+  list: filterList(state),
```

Comment:
list of what, can U use a more explicit name like dataflows or filteredDataflows ?